phpunit/blocks/render-block-navigation-test.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+class Render_Block_Navigation_Test extends WP_UnitTestCase {
```

Comment:
It might be useful to throw a third test together containing a group in to make sure nested & double nested submenus work. 

The current code is fine but it's always nice to protect ourselves from our future selves.

Something long winded such as:

```html
<!-- wp:group -->

<!-- wp:navigation-link {"label":"Sample Page","type":"page","id":2,"url":"http://localhost:8888/?page_id=2","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"Hello world!","type":"post","id":1,"url":"http://localhost:8888/?p=1","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-submenu {"label":"Uncategorized","type":"category","id":1,"url":"http://localhost:8888/?cat=1","kind":"taxonomy","isTopLevelItem":true} -->
<!-- wp:navigation-link {"label":"Sample Page","type":"page","id":3,"url":"http://localhost:8888/?page_id=3","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-submenu {"label":"Hello world!","type":"post","id":4,"url":"http://localhost:8888/?p=4","kind":"post-type","isTopLevelItem":false} -->
<!-- wp:navigation-link {"label":"Uncategorized","type":"category","id":5,"url":"http://localhost:8888/?cat=5","kind":"taxonomy","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"Hello world!","type":"post","id":6,"url":"http://localhost:8888/?p=6","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-submenu -->
<!-- /wp:navigation-submenu -->
<!-- /wp:group -->
```

Reply:
Tests in this case are an interesting one. I really don't know how this tests are going to be ported to core. It weirdly does feel right for the tests to live here in gutenberg plugin. Other blocks do not have unit tests, as far as I can see. If we are going to add unit tests for these functions, I think we should consider this when this functions live in core, where adding tests is much easier. I think the tests we have prove that the functions work but we should work on improving testing once in core.

Reply:
Unit test added.